tests/python/relay/test_vm_serialization.py

```diff
@@ -332,6 +284,20 @@ def test_mobilenet():
     run_network(mod, params)
 
 
+def test_vm_shape_of():
```

Comment:
I don't understand how this test hits the shape_of command? Are we sure this is testing the new code?

Reply:
`vm.shape_of` is implicitly added by memory manifest pass when a dynamic input is detected. It is used to get the shape of a tensor at runtime and it has to be on the CPU as the following instructions (i.e. `shape_func`) would be on CPU as well. The normal `shape_of` instruction doesn't have this constraint so we don't actually convert them.

https://github.com/apache/incubator-tvm/blob/ef9bf7d957665901c6980b9ada2aeb8e38d435c6/python/tvm/relay/transform/memory_alloc.py#L141

Reply:
:+1: Got it, thanks!